ocaml/sdk-gen/component-test/jsonrpc-client/go/main_test.go

```diff
@@ -50,3 +59,881 @@ func TestLoginSuccess(t *testing.T) {
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMImport(t *testing.T) {
+	session, err := getSession("test_id3")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var srRef xenapi.SRRef = "OpaqueRef:5ec2d621-7b62-f571-s38r-754341a973e5"
+	var VMImportURL string = "http://vm_import_url"
+	var result []xenapi.VMRef
+	var expectedResult = []xenapi.VMRef{"OpaqueRef:5ec2d621-7b62-f571-v38m-754341a973e5"}
+	result, err = xenapi.VM.Import(session, VMImportURL, srRef, false, false)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMRetrieveWlbRecommendations(t *testing.T) {
+	session, err := getSession("test_id4")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "OpaqueRef:5ec2d621-7b62-f571-v38m-754341a973e5"
+	var result map[xenapi.HostRef][]string
+	expectedResult := map[xenapi.HostRef][]string{
+		"OpaqueRef:4a0fde9c-5709-4f66-9968-64fb6162bd97": []string{
+			"host0",
+		},
+	}
+	result, err = xenapi.VM.RetrieveWlbRecommendations(session, vmRef)
+
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMMigrateSend(t *testing.T) {
+	session, err := getSession("test_id5")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "OpaqueRef:5vm2d621-7b62-f571-vm00-754341a973e5"
+	dest := map[string]string{
+		"name_label": "host1",
+	}
+	vdiMap := map[xenapi.VDIRef]xenapi.SRRef{
+		"OpaqueRef:5vdid621-7b62-f571-vdi8-754341a973e5": "OpaqueRef:5sr2d621-7b62-f571-s38r-754341a973e5",
+	}
+	vifMap := map[xenapi.VIFRef]xenapi.NetworkRef{
+		"OpaqueRef:9vifb8a9-f6a3-69d2-vifb-7a7b7c81c49e": "OpaqueRef:9network-f6a3-69d2-netw-7a7b7c81c49e",
+	}
+	var options map[string]string
+	vgpuMap := map[xenapi.VGPURef]xenapi.GPUGroupRef{
+		"OpaqueRef:9vgpu8a9-f6a3-69d2-vgpu-7a7b7c81c49e": "OpaqueRef:9gpugroup-f6a3-69d2-grou-7a7b7c81c49e",
+	}
+
+	var result xenapi.VMRef
+	var expectedResult xenapi.VMRef = "OpaqueRef:5vm2d621-7b62-f571-vm01-754341a973e5"
+	result, err = xenapi.VM.MigrateSend(session, vmRef, dest, true, vdiMap, vifMap, options, vgpuMap)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	if result != expectedResult {
+		t.Log("The result returned not the same with expected!")
+		t.Fail()
+		return
+	}
+
+	err = session.Logout()
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+}
+
+func TestVMGetRecord(t *testing.T) {
```

Comment:
I think we can remove this test case, it already been include in other test case.